### PR TITLE
feat: evaluate TEAL that uses template expressions

### PR DESCRIPTION
**Pull request by StylishTriangles**

### CODE REVIEW

File: tests/evaluator_tools_test.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+def test_compile_and_run_template():
+    program = Tmpl.Int("TMPL_A") + Tmpl.Int("TMPL_B")
+    subs = {
+        "TMPL_A": "123123",
+        "TMPL_B": "555555",
+    }
+    expected = 678678
+    stack, _ = compile_and_run(program, tmpl_subs=subs)
+
+    assert stack == [expected]
```
Comment (chwilko) [inline tests/evaluator_tools_test.py:54]: ok, but hypothesis seems better check edge cases

File: pytealext/evaluator/tools.py [hunk 2]
```diff
@@ -1,11 +1,60 @@
...
 def compile_and_run(
-    ast: Expr, mode=Mode.Application, version=MAX_PROGRAM_VERSION, *, context=None, debug=None
+    ast: Expr,
+    mode: Mode = Mode.Application,
+    version: int = MAX_PROGRAM_VERSION,
+    *,
+    context: EvalContext = None,
+    debug: IO = None,
+    tmpl_subs: dict[str, str] = None
 ) -> tuple[list[bytes or int], list[bytes or int]]:
```
Comment (chwilko) [inline pytealext/evaluator/tools.py:26]: why asterix in this place

Comment (StylishTriangles) [reply pytealext/evaluator/tools.py:26]: I think it would be hard to maintain this long list of positional arguments, also, the order wouldn't make sense. If you're asking why specifically this position, then it doesn't really make sense

The only good idea here is to keep mode positional, so it is the same as in compileTeal

Comment (StylishTriangles) [reply pytealext/evaluator/tools.py:26]: WDYT about moving the asterisk after mode?

**Review by chwilko — APPROVED** [review]
